metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_shares.py

```diff
@@ -45,64 +29,32 @@ def __init__(
         dataset_urn_builder: Callable[[str], str],
     ) -> None:
         self.config = config
-        self.shares = self.config.shares or {}
-        self.platform_instance = self.config.platform_instance or ""
         self.report = report
         self.logger = logger
         self.dataset_urn_builder = dataset_urn_builder
 
-    def _get_shared_databases(
-        self, shares: Dict[str, SnowflakeShareConfig], platform_instance: str
-    ) -> Dict[str, SharedDatabase]:
-        shared_databases: Dict[str, SharedDatabase] = {}
-
-        for share_name, share_details in shares.items():
-            if share_details.platform_instance == platform_instance:
-                if share_details.database not in shared_databases:
-                    shared_databases[share_details.database] = SharedDatabase(
-                        name=share_details.database,
-                        created_from_share=False,
-                        shares=[share_name],
-                    )
-
-                else:
-                    shared_databases[share_details.database].shares.append(share_name)
-
-            else:
-                for consumer in share_details.consumers:
-                    if consumer.platform_instance == platform_instance:
-                        shared_databases[consumer.database] = SharedDatabase(
-                            name=share_details.database,
-                            created_from_share=True,
-                            shares=[share_name],
-                        )
-                        break
-                else:
-                    self.report_warning(
-                        f"Skipping Share, as it does not include current platform instance {platform_instance}",
-                        share_name,
-                    )
-
-        return shared_databases
-
     def get_shares_workunits(
         self, databases: List[SnowflakeDatabase]
     ) -> Iterable[MetadataWorkUnit]:
-        shared_databases = self._get_shared_databases(
-            self.shares, self.platform_instance
-        )
-
+        inbounds = self.config.inbounds()
+        outbounds = self.config.outbounds()
         # None of the databases are shared
-        if not shared_databases:
+        if not (inbounds or outbounds):
             return
 
         logger.debug("Checking databases for inbound or outbound shares.")
         for db in databases:
-            if db.name not in shared_databases:
+            db.name = db.name
+            is_inbound = db.name in inbounds
+            is_outbound = db.name in outbounds
+
+            if not (is_inbound or is_outbound):
                 logger.debug(f"database {db.name} is not shared.")
                 continue
 
-            sibling_dbs = self.get_sibling_databases(shared_databases[db.name])
+            sibling_dbs = (
```

Comment:
If this is typed as `Collection` or `Iterable` then you don't have to cast to list. Doesn't matter though